roachtest: ycsb/E/nodes=3/cpu=32 failed

SHA: https://github.com/cockroachdb/cockroach/commits/c11a254ed74c59ac3ff083356d9a71b82db2c7d7

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=ycsb/E/nodes=3/cpu=32 PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1505083&tab=artifacts#/ycsb/E/nodes=3/cpu=32

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20190925-1505083/ycsb/E/nodes=3/cpu=32/run_1
	cluster.go:1762,ycsb.go:36,cluster.go:2118,errgroup.go:57: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1569387937-55-n4cpu32:4 -- ./workload run ycsb --init --insert-count=1000000 --splits=100 --workload=E --concurrency=64 --histograms=perf/stats.json --ramp=1m --duration=10m {pgurl:1-3} returned:
		stderr:
		
		stdout:
		)
			/usr/local/go/src/runtime/proc.go:301 +0xef fp=0xc000a685f8 sp=0xc000a685d8 pc=0x43a39f
		runtime.selectgo(0xc000a68758, 0xc000a68748, 0x2, 0x0, 0x0)
			/usr/local/go/src/runtime/select.go:313 +0xcb8 fp=0xc000a68720 sp=0xc000a685f8 pc=0x449e78
		database/sql.(*Rows).awaitDone(0xc000dbc480, 0x1c049a0, 0xc0009ac700, 0x0, 0x0)
			/usr/local/go/src/database/sql/sql.go:2642 +0xcc fp=0xc000a687b8 sp=0xc000a68720 pc=0x50864c
		runtime.goexit()
			/usr/local/go/src/runtime/asm_amd64.s:1337 +0x1 fp=0xc000a687c0 sp=0xc000a687b8 pc=0x467a31
		created by database/sql.(*Rows).initContextClose
			/usr/local/go/src/database/sql/sql.go:2630 +0xc1
		bash: line 1: 22453 Aborted                 (core dumped) bash -c "./workload run ycsb --init --insert-count=1000000 --splits=100 --workload=E --concurrency=64 --histograms=perf/stats.json --ramp=1m --duration=10m 'postgres://root@10.128.0.195:26257?sslmode=disable' 'postgres://root@10.128.0.194:26257?sslmode=disable' 'postgres://root@10.128.0.193:26257?sslmode=disable'"
		Error:  exit status 134
		: exit status 1

```

It appears that the YCSB workload crashed.

```
panic: runtime error: index out of range

goroutine 1673 [running]:
panic(0x1548e40, 0x2a539c0)
	/usr/local/go/src/runtime/panic.go:565 +0x2c5 fp=0xc001230c20 sp=0xc001230b90 pc=0x438285
runtime.panicindex()
	/usr/local/go/src/runtime/panic.go:44 +0xfb fp=0xc001230c50 sp=0xc001230c20 pc=0x43712b
math/rand.(*rngSource).Uint64(...)
	/usr/local/go/src/math/rand/rng.go:249
math/rand.(*rngSource).Int63(0xc000202a00, 0x3)
	/usr/local/go/src/math/rand/rng.go:234 +0x8a fp=0xc001230c60 sp=0xc001230c50 pc=0x50ee0a
math/rand.(*Rand).Int63(...)
	/usr/local/go/src/math/rand/rand.go:85
math/rand.(*Rand).Float64(...)
	/usr/local/go/src/math/rand/rand.go:196
github.com/cockroachdb/cockroach/pkg/workload/ycsb.(*ZipfGenerator).Uint64(0xc00106b9e0, 0x11f9ab)
	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/workload/ycsb/zipfgenerator.go:132 +0x6d fp=0xc001230d18 sp=0xc001230c60 pc=0x129376d
github.com/cockroachdb/cockroach/pkg/workload/ycsb.(*ycsbWorker).nextReadKey(0xc001135bc0, 0x43106a7b4f1, 0x35eb7b58)
	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/workload/ycsb/ycsb.go:481 +0x57 fp=0xc001230d50 sp=0xc001230d18 pc=0x1291ec7
github.com/cockroachdb/cockroach/pkg/workload/ycsb.(*ycsbWorker).scanRows(0xc001135bc0, 0x1c049a0, 0xc0016bb080, 0x0, 0x0)
	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/workload/ycsb/ycsb.go:557 +0x4b fp=0xc001230df0 sp=0xc001230d50 pc=0x12927ab
github.com/cockroachdb/cockroach/pkg/workload/ycsb.(*ycsbWorker).run(0xc001135bc0, 0x1c049a0, 0xc0016bb080, 0x0, 0xc0016bb090)
	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/workload/ycsb/ycsb.go:416 +0x156 fp=0xc001230e78 sp=0xc001230df0 pc=0x12919d6
github.com/cockroachdb/cockroach/pkg/workload/ycsb.(*ycsbWorker).run-fm(0x1c049a0, 0xc0016bb080, 0x0, 0x0)
	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/workload/ycsb/ycsb.go:403 +0x3e fp=0xc001230eb0 sp=0xc001230e78 pc=0x129527e
github.com/cockroachdb/cockroach/pkg/workload/cli.workerRun(0x1c049a0, 0xc0016bb080, 0xc0008074a0, 0xc0002e1370, 0x0, 0xc00035bac0)
	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/workload/cli/run.go:235 +0x94 fp=0xc001230f20 sp=0xc001230eb0 pc=0x1346444
github.com/cockroachdb/cockroach/pkg/workload/cli.runRun.func1.1(0xc0002d6000, 0x170894c, 0x4, 0xc000310200, 0x40, 0x40, 0x0, 0x0, 0x0, 0xc0008074a0, ...)
	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/workload/cli/run.go:401 +0xfd fp=0xc001230f60 sp=0xc001230f20 pc=0x1349ebd
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1337 +0x1 fp=0xc001230f68 sp=0xc001230f60 pc=0x467a31
created by github.com/cockroachdb/cockroach/pkg/workload/cli.runRun.func1
	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/workload/cli/run.go:391 +0x133
```

@nvanbenschoten is this you?

--- a/pkg/workload/ycsb/ycsb.go
+++ b/pkg/workload/ycsb/ycsb.go
@@ -307,21 +307,21 @@ func (g *ycsb) Ops(urls []string, reg *histogram.Registry) (workload.QueryLoad, 
 		}
 	}
 
-	zipfRng := rand.New(rand.NewSource(g.seed))
-	var requestGen randGenerator
-	var scanLengthGen randGenerator
-	var rowIndex = new(uint64)
-	*rowIndex = uint64(g.recordCount)
+	rowIndexVal := uint64(g.recordCount)
+	rowIndex := &rowIndexVal
+	rowCounter := NewAcknowledgedCounter((uint64)(g.recordCount))
 
+	var requestGen randGenerator
+	requestGenRng := rand.New(rand.NewSource(g.seed))
 	switch strings.ToLower(g.requestDistribution) {
 	case "zipfian":
 		requestGen, err = NewZipfGenerator(
-			zipfRng, zipfIMin, defaultIMax-1, defaultTheta, false /* verbose */)
+			requestGenRng, zipfIMin, defaultIMax-1, defaultTheta, false /* verbose */)
 	case "uniform":
-		requestGen, err = NewUniformGenerator(zipfRng, 0, uint64(g.recordCount)-1)
+		requestGen, err = NewUniformGenerator(requestGenRng, 0, uint64(g.recordCount)-1)
 	case "latest":
 		requestGen, err = NewSkewedLatestGenerator(
-			zipfRng, zipfIMin, uint64(g.recordCount)-1, defaultTheta, false /* verbose */)
+			requestGenRng, zipfIMin, uint64(g.recordCount)-1, defaultTheta, false /* verbose */)
 	default:
 		return workload.QueryLoad{}, errors.Errorf("Unknown request distribution: %s", g.requestDistribution)
 	}
@@ -329,11 +329,13 @@ func (g *ycsb) Ops(urls []string, reg *histogram.Registry) (workload.QueryLoad, 
 		return workload.QueryLoad{}, err
 	}
 
+	var scanLengthGen randGenerator
+	scanLengthGenRng := rand.New(rand.NewSource(g.seed + 1))
 	switch strings.ToLower(g.scanLengthDistribution) {
 	case "zipfian":
-		scanLengthGen, err = NewZipfGenerator(zipfRng, g.minScanLength, g.maxScanLength, defaultTheta, false /* verbose */)
+		scanLengthGen, err = NewZipfGenerator(scanLengthGenRng, g.minScanLength, g.maxScanLength, defaultTheta, false /* verbose */)
 	case "uniform":
-		scanLengthGen, err = NewUniformGenerator(zipfRng, g.minScanLength, g.maxScanLength)
+		scanLengthGen, err = NewUniformGenerator(scanLengthGenRng, g.minScanLength, g.maxScanLength)
 	default:
 		return workload.QueryLoad{}, errors.Errorf("Unknown scan length distribution: %s", g.scanLengthDistribution)
 	}
@@ -341,7 +343,6 @@ func (g *ycsb) Ops(urls []string, reg *histogram.Registry) (workload.QueryLoad, 
 		return workload.QueryLoad{}, err
 	}
 
-	rowCounter := NewAcknowledgedCounter((uint64)(g.recordCount))
 	ql := workload.QueryLoad{SQLDatabase: sqlDatabase}
 	for i := 0; i < g.connFlags.Concurrency; i++ {
 		rng := rand.New(rand.NewSource(g.seed + int64(i)))

